Doc/conf.py

```diff
@@ -290,7 +290,7 @@
 
 # If not '', a 'Last updated on:' timestamp is inserted at every page bottom,
 # using the given strftime format.
-html_last_updated_fmt = '%b %d, %Y'
+html_last_updated_fmt = '%b %d, %Y (%H:%M, GMT%z)'
```

Comment:
OK, thanks! 

I think ideally Sphinx should use UTC instead of local.

But given it's local, and the RtD preview server is UTC, the docs server is likely UTC, a local build is the timezone of your machine, so we should include the offset.

And I'd suggest "UTC" rather than "GMT"?

```suggestion
html_last_updated_fmt = '%b %d, %Y (%H:%M UTC%z)'
```

Reply:
Interestingly, [virtualenv uses](https://github.com/pypa/virtualenv/blob/fc5627ea281ca3074e072fd7fc7d831a976778f1/docs/conf.py#L34C13-L34C34):

```python
html_last_updated_fmt = datetime.now(tz=timezone.utc).isoformat()
```

Yep, just put your own timestamp there with no percent formatters! For [example](https://virtualenv.pypa.io/en/latest/):

> Last updated on 2023-09-26T22:37:21.834119+00:00

So we could do something like this, and like [Wikipedia](https://en.wikipedia.org/wiki/Python_(programming_language)), omit "UTC":

```suggestion
import datetime as dt
html_last_updated_fmt = dt.datetime.utcnow().strftime('%b %d, %Y (%H:%M)')
```

> Last updated on Sep 30, 2023 (09:26).

Reply:
> So we could do something like this, and like Wikipedia, omit "UTC"

I'm not sure what you're talking about, Wikipedia shows "UTC" to me:

> This page was last edited on 28 September 2023, at 04:53 (**UTC**).

Please. Add the timezone. Please. Use UTC timezone if possible.

I like the approach of omitting "%" in html_last_updated_fmt and just use a string to override the timestamp created by Sphinx.

Reply:
Ah interesting, I'm logged in to Wikipedia and it shows:

> This page was last edited on 28 September 2023, at 04:53.

With incognito, it shows:

> This page was last edited on 28 September 2023, at 04:53 (UTC).

I don't remember changing any settings :)

---

Anyway, this includes "UTC", and of course move the import to the top:

```suggestion
import datetime as dt
html_last_updated_fmt = dt.datetime.utcnow().strftime('%b %d, %Y (%H:%M UTC)')
```

> Last updated on Sep 30, 2023 (11:05 UTC).

Reply:
"With incognito, it shows:"

Maybe you find a mode to time travel?

Reply:
> I think ideally Sphinx should use UTC instead of local.
> 
> But given it's local, and the RtD preview server is UTC, the docs server is likely UTC, a local build is the timezone of your machine, so we should include the offset.

Interestingly, I would expect the time to be in local time here -- e.g. if it said "Last updated on Oct 1, 2023 (10:00)", but it was 11 O'Clock, I'd be surprised. Adding "GMT" (UTC, UT1, etc) would help, but I'd still need to do the timezone maths in my head to work it out. I think putting the local time with the offset from the meridian makes the most sense, as really the only reason we're including the time in the first place is to enable us working out when the expected time to the *next* re-build is.

A

Reply:
Where is "here"? I don't know where your live, I don't know when your timezone. Same for the server building the doc.

Reply:
Yeah, that makes sense.

For something like an online conference programme, it's helpful to use JavaScript to show the times in the viewer's own timezone.

For a timestamp such as this, static time+timezone is just fine.